File: backend/app/models/system.py
from pydantic import BaseModel, Field


class SystemStatus(BaseModel):
    cpu_percent: float | None = Field(default=None, alias="cpuPercent")
    memory_percent: float | None = Field(default=None, alias="memoryPercent")
    gpu_percent: float | None = Field(default=None, alias="gpuPercent")
    network: str = "ONLINE"
    timestamp: float

    model_config = {"populate_by_name": True}

File: frontend/src/types/system.ts
export const SubsystemStatus = {
  ONLINE: 'ONLINE',
  STANDBY: 'STANDBY',
  OFFLINE: 'OFFLINE',
  ERROR: 'ERROR',
  ACTIVE: 'ACTIVE',
} as const

export type SubsystemStatus = (typeof SubsystemStatus)[keyof typeof SubsystemStatus]

export interface SystemStatusPayload {
  cpuPercent: number | null
  memoryPercent: number | null
  gpuPercent: number | null
  network: SubsystemStatus
  timestamp: number
}

export interface SubsystemMap {
  vision: SubsystemStatus
  voice: SubsystemStatus
  ai: SubsystemStatus
  spatial: SubsystemStatus
  network: SubsystemStatus
  camera: SubsystemStatus
  microphone: SubsystemStatus
}
